src/openvic-simulation/politics/NationalFocus.hpp

```diff
@@ -0,0 +1,76 @@
+#pragma once
+
+#include "openvic-simulation/types/IdentifierRegistry.hpp"
+#include "openvic-simulation/utility/Getters.hpp"
+#include "openvic-simulation/Modifier.hpp"
+#include "openvic-simulation/pop/Pop.hpp"
+#include "openvic-simulation/politics/Ideology.hpp"
+#include "openvic-simulation/economy/Good.hpp"
+
+#include <optional>
+
+namespace OpenVic {
+	struct NationalFocusManager;
+
+	struct NationalFocusGroup : HasIdentifier {
+		friend struct NationalFocusManager;
+	
+	private:
+		NationalFocusGroup(std::string_view new_identifier);
+	};
+
+	struct NationalFocus : HasIdentifier {
+		friend struct NationalFocusManager;
+
+	public:
+		using pop_promotion_map_t = std::map<PopType const*, fixed_point_t>;
+		using party_loyalty_map_t = std::map<Ideology const*, fixed_point_t>;
+		using production_map_t = std::map<Good const*, fixed_point_t>;
+
+	private:
+		std::string PROPERTY(icon);
```

Comment:
This should be a uint*_t (width is up to you), and check in add_national_focus should be that it's non-zero (the 0-th icon is the "no/unset national focus" one)